lib/pure/math.nim

```diff
@@ -157,41 +157,38 @@ func isNaN*(x: SomeFloat): bool {.inline, since: (1,5,1).} =
     else: result = c_isnan(x)
 
 func copySign*[T: SomeFloat](x, y: T): T {.inline, since: (1, 5, 1).} =
-  ## Returns a value with the magnitude of `x` and the sign of `y`.
+  ## Returns a value with the magnitude of `x` and the sign of `y`;
+  ## this works even if x or y are NaN or zero, both of which can carry a sign.
   runnableExamples:
-    doAssert copySign(10.0, -1.0) == -10.0
-    doAssert copySign(-10.0, -1.0) == -10.0
-    doAssert copySign(-10.0, 1.0) == 10.0
-    doAssert copySign(10'f32, -1.0) == -10.0
-
-    doAssert copySign(Inf, -1.0) == -Inf
-    doAssert copySign(-Inf, 1.0) == Inf
     doAssert copySign(1.0, -0.0) == -1.0
     doAssert copySign(0.0, -0.0) == -0.0
     doAssert copySign(-1.0, 0.0) == 1.0
     doAssert copySign(10.0, 0.0) == 10.0
+
+    doAssert copySign(Inf, -1.0) == -Inf
+    doAssert copySign(-Inf, 1.0) == Inf
     doAssert copySign(-1.0, NaN) == 1.0
     doAssert copySign(10.0, NaN) == 10.0
 
     doAssert copySign(NaN, 0.0).isNaN
     doAssert copySign(NaN, -0.0).isNaN
+
+    # fails in VM and JS backend
```

Comment:
it won't fail in VM for a nim binary compiled after your PR

Reply:
The same story as https://github.com/nim-lang/Nim/pull/16406#discussion_r549608120